editor.planx.uk/src/components/EditorNavMenu/EditorNavMenu.tsx

```diff
@@ -24,15 +25,16 @@ function EditorNavMenu() {
   const { navigate } = useNavigation();
   const { url } = useCurrentRoute();
   const isRouteLoading = useLoadingRoute();
-  const [teamSlug, flowSlug, flowAnalyticsLink, role, flowId] = useStore(
+  const [teamSlug, flowSlug, flowAnalyticsLink, role, team] = useStore(
     (state) => [
       state.teamSlug,
       state.flowSlug,
       state.flowAnalyticsLink,
       state.getUserRoleForCurrentTeam(),
-      state.id,
+      state.getTeam(),
     ],
   );
+  const referenceCode = team?.settings?.referenceCode;
```

Comment:
There might be a bit of confusion here between `referenceCode` & `hasPlanningData` which is a manual toggle - I _think_ that overall we're headed towards a future where `hasPlanningData` goes away and configuring your reference code alone via Team Settings page could be enough to "enable" constraint checks? One to discuss!